### PR TITLE
fix(Core/DB/Events): Game Event Start Date & Events on Calendar

### PR DESCRIPTION
**Pull request by Kitzunu**
https://github.com/TrinityCore/TrinityCore/commit/299d6ebbffd11a8fa92327c5a9296602a1596202
https://github.com/TrinityCore/TrinityCore/commit/87c79c9485cbb9536912e1566174c717946c1537
https://github.com/TrinityCore/TrinityCore/commit/4913edf81bfbd505e3218161b45705962876552d
https://github.com/TrinityCore/TrinityCore/commit/f41dfa0089a7ba8719f28816ae75ba91313df5c4

## CHANGES PROPOSED:
-  Implement holiday_duration in holiday_dates (holiday_duration is the number of hours of how long the holiday may last (visible in the calendar))
-  Update Start and End dates of events

## ISSUES ADDRESSED:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/2550
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/542

## TESTS PERFORMED:
- Builds
- Checked in-game calendar and start dates. All bueno

## HOW TO TEST THE CHANGES:

1. Check calendar that all event frames are shown
2. .ev #event_id (To check start times)

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by merdlin** [comment]
I did the test and it works, thanks.

**Comment by FrancescoBorzi** [comment]
thanks @Kitzunu  and @merdlin

File: data/sql/updates/pending_db_world/rev_1586845118268975400.sql [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+(@HOLIDAYID, @DATEID+4, 411484160), -- 2024-09-28.
+(@HOLIDAYID, @DATEID+5, 428261376); -- 2025-09-28.
+
+-- Lunar Festival
+UPDATE `game_event` SET `length` = 20160 WHERE `eventEntry` = 7;
+UPDATE `holiday_dates` SET `holiday_duration`=336 WHERE `id`=327;
+
+-- End Dates
+
+UPDATE `game_event` SET `end_time` = '2030-12-31 06:00:00' WHERE `eventEntry` IN (3,  73,  81,  79,  53,  54,  77,  70,  80,  71,  65,  34,  35,  64,  63,  62,  43,  51,  36,  37,  69,  68,  67,  45,  44,  42,  41,  40,  39,  38,  52,  26,  27,  25,  24,  11,  12,  23,  21,  20,  19,  18,  14,  10,  9,  1,  2,  4,  30,  5,  15,  6,  29,  7,  8,  28,  16,  50,  72,  75,  74,  33,  76,  32,  78);
```
Comment (LichKing255) [inline data/sql/updates/pending_db_world/rev_1586845118268975400.sql:94]: We don't need to set End Time. delete end_time lines.

https://github.com/azerothcore/azerothcore-wotlk/blob/c839ce53988c544052ae96a82f325fe441ad1747/src/server/game/Events/GameEventMgr.cpp#L244

Comment (FrancescoBorzi) [reply data/sql/updates/pending_db_world/rev_1586845118268975400.sql:94]: @Kitzunu

File: data/sql/updates/pending_db_world/rev_1586845118268975400.sql [hunk 2]
```diff
@@ -0,0 +1,90 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1586845118268975400');
+
+ALTER TABLE `holiday_dates` ADD COLUMN `holiday_duration` INT(10) UNSIGNED DEFAULT 0 NOT NULL AFTER `date_value`;
+UPDATE `holiday_dates` SET `holiday_duration`= 360 WHERE `id`= 327;
+
+-- Darkmoon Faire
+UPDATE `game_event` SET `start_time`= '2020-01-05 00:01:00', `end_time`= '2030-01-01 06:00:00' WHERE `eventEntry`= 3;
```
Comment (LichKing255) [inline data/sql/updates/pending_db_world/rev_1586845118268975400.sql:7]: and here delete ``, `end_time`= '2030-01-01 06:00:00'``

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1586845118268975400.sql:7]: Ah totally missed it, will fix later

**Comment by Kitzunu** [comment]
Thank you @merdlin and @LichKing255 for test & review :)

**Comment by BarbzYHOOL** [comment]
nice

**Comment by sudlud** [comment]
so we've reached the end of these dates now with this years' winter veil. so time has come for a new pr so there will be game events in the in game calendar for 2026+ :D

**Pull request merged**